### ISSUE TITLE
services not available during module installation

### ISSUE THREAD

**Issue by Wiatrogon** [PrestaShop/PrestaShop#19627]
I'm not really sure if this is a bug or just omission in documentation.
I tried to create a module that installs a tab as shown in
https://devdocs.prestashop.com/1.7/modules/concepts/controllers/admin-controllers/tabs/#modern-controllers-and-manual-tab-insertion
Of course I thought "lets make a TabInstaller service" since I've heard that SOLID is a thing.
Shortly after I've had a `install()` method in my module that looked somewhat like this:
``` // mymodule.php
    public function install()
    {
        return parent::install()
            && $this->get('my_module.tab_installer')->install();
    }
```
and my `config/admin/services.yml` contained something like this:
```
services:
    my_module.tab_installer:
        class: MyModule\TabInstaller
```
Then, after updating my module files and clearing cache I tried to install my module and got `You have requested a non-existent service`

I have several questions now:
* is this a bug or services are just unavailable in `Module::install()`?
* are services available in Module in places other than `getContent` and/or hooks?
* should I create an issue in PrestaShop/docs for them to state explicitly when are services defined by custom module available to get from container?

Regards

**Comment by sarahdib** [comment]
Hello, 

Please be aware that this bug tracker should be used to report issues only. 
If you need help, you can ask on the :
- Forum (https://www.prestashop.com/forums/)
- Stack Overflow (https://stackoverflow.com/questions/tagged/prestashop)

You can also join slack in order to discuss with developer https://github.com/PrestaShop/open-source/blob/master/slack/readme.md

Thanks for your understanding!

**Comment by sarahdib** [comment]
@PrestaShop/prestashop-core-developers Could you check ?

**Comment by jolelievre** [comment]
Hello @Wiatrogon 

Thank you for your feedback, this is indeed an issue that we noticed An issue has already been created for this https://github.com/PrestaShop/PrestaShop/issues/14527 It goes beyond services only, there are also limitations with Doctrine entities and even autoload files. The limitation is only during the install phase, because the PrestaShop core only scans for services and doctrine entities from **installed** modules, so when the install process starts it is not yet installed..

I started a PR to fix those issues https://github.com/PrestaShop/PrestaShop/pull/17706, it is on standby for now because it ended up being more complex regarding the service loading. But we think we found a solution by rebooting the kernel during installation (there is an ADR that details this decision https://github.com/PrestaShop/ADR/pull/8)

Anyway this bug will be fixed in 1.7.8 (it was too late to finish it and include it in 1.7.7). But I'm gonna check the doc, as you said, because this requires some clarification.

I hope I answered all your questions? Feel free to ask more if you have other ones. I'm gonna create a PR to update the doc and then I will close this issue as it is a duplicate of #14527 :)
